compute_tools/src/migration.rs

```diff
@@ -81,53 +77,50 @@ impl<'m> MigrationRunner<'m> {
         Ok(())
     }
 
-    /// Run the configrured set of migrations
-    pub fn run_migrations(mut self) -> Result<()> {
-        self.prepare_database()?;
+    /// Run an individual migration
+    fn run_migration(txn: &mut Transaction, migration_id: i64, migration: &str) -> Result<()> {
+        if migration.starts_with("-- SKIP") {
+            info!("Skipping migration id={}", migration_id);
 
-        let mut current_migration = self.get_migration_id()? as usize;
-        while current_migration < self.migrations.len() {
-            macro_rules! migration_id {
-                ($cm:expr) => {
-                    ($cm + 1) as i64
-                };
-            }
+            // Even though we are skipping the migration, updating the
+            // migration ID should help keep logic easy to understand when
+            // trying to understand the state of a cluster.
+            Self::update_migration_id(txn, migration_id)?;
+        } else {
+            info!("Running migration id={}:\n{}\n", migration_id, migration);
 
-            let migration = self.migrations[current_migration];
+            txn.simple_query(migration)
+                .with_context(|| format!("apply migration id={migration_id}"))?;
 
-            if migration.starts_with("-- SKIP") {
-                info!("Skipping migration id={}", migration_id!(current_migration));
+            Self::update_migration_id(txn, migration_id)?;
 
-                // Even though we are skipping the migration, updating the
-                // migration ID should help keep logic easy to understand when
-                // trying to understand the state of a cluster.
-                self.update_migration_id(migration_id!(current_migration))?;
-            } else {
-                info!(
-                    "Running migration id={}:\n{}\n",
-                    migration_id!(current_migration),
-                    migration
-                );
+            info!("Finished migration id={}", migration_id);
```

Comment:
Not sure, but I'd probably move this message after `commit()` inside `run_migrations()`, so that 'Finished' was printed only after commit. It's fine to print it in case of Skip, because we i) log Skip message; ii) even when we skip we still modify the migration id.

Reply:
Good find here. Makes a ton of sense.